packages/gatsby/src/utils/app-data.ts

```diff
@@ -1,18 +1,12 @@
-const fs = require(`fs-extra`)
-const path = require(`path`)
+import fs from "fs-extra"
+import path from "path"
 
 const APP_DATA_JSON = `app-data.json`
 
-const write = (publicDir, hash) => {
+export const write = (publicDir: string, hash: string): Promise<void> =>
```

Comment:
I changed to  returned Promise  because it is awaiting in build.js

https://github.com/gatsbyjs/gatsby/blob/e6696d3661ca0b8e9357a1b107428296299e739f/packages/gatsby/src/commands/build.js#L138

Reply:
You can await a normal function too, so no need to make this return a Promise here, since it’s not async

Reply:
Sorry, Fix it.

https://github.com/gatsbyjs/gatsby/pull/22076/commits/b56c12727dc1a68f2e792a92d3bd4ca28f986f60